src/include/duckdb/common/unique_ptr.hpp

```diff
@@ -53,38 +73,11 @@ class unique_ptr : public std::unique_ptr<_Tp, _Dp> {
 	}
 };
 
-template <class _Tp, class _Dp, bool SAFE>
```

Comment:
I agree with all changes except for consolidating the `unique_ptr[]` together with the `unique_ptr`

I believe the idea we had for this was to eventually remove `unique_ptr[]` altogether and move it into it's own class, `unique_array`, hence the `using` at the bottom of this file.

I think combining the unique array features into the unique_ptr will complicate this move, and we now also have a potential issue because you can use the `[]` operator on a regular unique_ptr

Reply:
I agree its not ideal, I considered adding a static_assertion on the operator[] to catch at compile time uses from the non-array class, potentially that could work, unsure.

Current status one minor but hard to catch problem is that stuff like `unique_ptr<A[], some_deleter>` I guess will get the non-array treatment (than likely failing on `operator[]`, or maybe not since access might be after a cast / something else).

Or maybe the inverted reasoning that could make more sense: keep the classes as now, but add a static_assertion on unique_ptr<A> being used only from non-arrays.

I am not sure of the way out.

Reply:
Let's leave the unique pointer alone and have this PR only fix the `shared_ptr` issues?

Reply:
Unfortunately shared_ptr and unique_ptr are a bit tangled together, tomororw I will try to see if there is a decent way out, otherwise we can also just stash it until there is some stronger motivation to take a fresh look here.

Reply:
Actually there was a commit that was easily separable, filed that in a separate PR.